File: solutions/387-first-unique-character-in-a-string/first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> firstUnique;
        string letters = "abcdefghijklmnopqrstuvwxyz";
        
        for (char letter : letters) {
            int count = 0;
            for (char c : s) {
                if (c == letter) count++;
            }
            if (count == 1) {
                firstUnique.push_back(s.find(letter));
            }
        }
        
        return firstUnique.empty() ? -1 : *min_element(firstUnique.begin(), firstUnique.end());
    }
};

File: solutions/387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        ArrayList<Integer> firstUnique = new ArrayList<>();
        String letters = "abcdefghijklmnopqrstuvwxyz";
        
        for (char letter : letters.toCharArray()) {
            long count = s.chars().filter(ch -> ch == letter).count();
            if (count == 1) {
                firstUnique.add(s.indexOf(letter));
            }
        }
        
        return firstUnique.isEmpty() ? -1 : Collections.min(firstUnique);
    }
}